lib/declare_schema/model.rb

```diff
@@ -145,7 +149,7 @@ def belongs_to(name, scope = nil, **options, &block)
           index([foreign_type, fkey], index_options) if index_options[:name] != false
         else
           index(fkey, index_options) if index_options[:name] != false
-          options[:constraint_name] = options
+          # options[:constraint_name] = options
```

Comment:
This is bizarre to point `options` at itself! It was leading to infinite recursion in my test. I hope we can live without it.

Reply:
Ha, I must've loaded the PR before you left this comment. 🔪